File: src/ios/DecibelPlugin.swift
/********* DecibelPlugin.swift Cordova Plugin Implementation *******/

import Foundation

@objc(DecibelPlugin)
class DecibelPlugin : CDVPlugin {
    @objc(startListening:)func startListening(_ command: CDVInvokedUrlCommand) {
        self.commandDelegate!.send(CDVPluginResult(status: CDVCommandStatus_OK), callbackId: command.callbackId)
    }

    @objc(stopListening:)func stopListening(_ command: CDVInvokedUrlCommand) {
        self.commandDelegate!.send(CDVPluginResult(status: CDVCommandStatus_OK), callbackId: command.callbackId)
    }
}

File: www/DecibelPlugin.js
var exec = require('cordova/exec');

exports.startListening = function (decibelResult, error, pollingFrequency = 50) {
    exec(decibelResult, error, 'DecibelPlugin', 'startListening', [pollingFrequency]);
};

exports.stopListening = function (success, error) {
    exec(success, error, 'DecibelPlugin', 'stopListening', []);
};
